### PR TITLE
Add RAII wrapper for Win32 IMM context

### PR DESCRIPTION
**Pull request by cbracken**
Wraps Win32 IME context management in an RAII container that automates
management via ImmGetContext() and ImmReleaseContext().

Fixup for TODO left behind in flutter/engine#23853.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by cbracken** [comment]
Re-run of Mac Host Engine: https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20Host%20Engine/21155/overview

**Comment by cbracken** [comment]
Re: tests. This simply refactors the existing IMM context creation/cleanup.

**Review by clarkezone — APPROVED** [review]
LGTM

File: shell/platform/windows/text_input_manager_win32.cc [hunk 1]
```diff
@@ -10,10 +10,44 @@
...
+class ImmContext {
+ public:
+  ImmContext(HWND window_handle)
+      : context_(::ImmGetContext(window_handle)),
+        window_handle_(window_handle) {
+    assert(window_handle);
+  }
+
+  ~ImmContext() {
+    if (window_handle_ != nullptr && context_ != nullptr) {
```
Comment (stuartmorgan) [inline shell/platform/windows/text_input_manager_win32.cc:28]: Why are you checking `window_handle_` when it's asserted to be non-null at construction?

**Review by stuartmorgan — APPROVED** [review]
LGTM with nit.

File: shell/platform/windows/text_input_manager_win32.cc (hunk 1 above)
Comment (cbracken) [reply shell/platform/windows/text_input_manager_win32.cc]: Good question! Added the assert at the last minute and forgot to remove the check from the dtor. Fixed.

**Comment by cbracken** [comment]
Once we extract the `UniqueObject` template from FML, we could likely replace this code with an instantiation of that.

**Pull request merged**